.circleci/config.yml

```
@@ -0,0 +1,82 @@
+version: 2.1
+
+executors:
+  python:
+    docker:
+      - image: cimg/python:3.11.2
+    resource_class: small
+
+filter-not-main: &filter-not-main
+  filters:
+    branches:
+      ignore:
+        - main
+        - /^dependabot/(?!docker/).*/
+        - /^hotfix\/.+/
+
+filter-main: &filter-main
+  filters:
+    branches:
+      only:
+        - main
+        - /^hotfix\/.+/
+
+orbs:
+  serverless-framework: circleci/serverless-framework@2.0.1
+  slack: circleci/slack@4.3.0
+
+jobs:
+  check:
+    executor: python
+    steps:
+      - checkout
+      - run: |
+          python -m venv venv
+          source venv/bin/activate
+          pip install ruff
+      - run: make check
```

```suggestion
      - run: |
          python -m venv venv
          source venv/bin/activate
          pip install ruff
          make check
```

You'll want that on the same block otherwise CircleCi will treat it as a new step and forget everything it ever knew about the last one